### PR TITLE
Goalui

### PR DESCRIPTION
**Pull request by ciliakas**
Feature updates:
- Added saving type selection to expanded goal form
- Added a label stating if the user can save enough or not
- Improved switching between goals

### CODE REVIEW

File: SmartSaver/forms/finances/goals/ExpandedGoalForm.cs [hunk 1]
```diff
@@ -50,21 +54,34 @@ private void Init()
...
             dataGridView.DataSource = _handler.DataTableConverter.GenerateSuggestionTable(entrySuggestions);
 
             dataGridView.Columns["ID"].Visible = false;
             dataGridView.Columns["Amount"].DefaultCellStyle.Format = "c";
             dataGridView.Columns["Date"].DefaultCellStyle.Format = "d";
         }
+
+        private void SetComboBoxSelections()
+        {
+            object[] names = Enum.GetNames(typeof(SavingType));
```
Comment (Stokalas) [inline SmartSaver/forms/finances/goals/ExpandedGoalForm.cs:76]: why is this not var?

**Review by Stokalas — APPROVED** [review]
Nothing much to say codewise, looks good, just one question.

File: SmartSaver/forms/finances/goals/ExpandedGoalForm.cs (hunk 1 above)
Comment (ciliakas) [reply SmartSaver/forms/finances/goals/ExpandedGoalForm.cs:76]: Reshaper suggested that there might be errors when converting in run time strings to objects, so in this way we avoid that error.

**Review by MatasLa — APPROVED** [review]
Looks good. Great job again.

**Pull request merged**